glide-core/src/client/mod.rs

```diff
@@ -204,6 +204,7 @@ fn get_request_timeout(cmd: &Cmd, default_timeout: Duration) -> RedisResult<Opti
             .position(b"BLOCK")
             .map(|idx| get_timeout_from_cmd_arg(cmd, idx + 1, TimeUnit::Milliseconds))
             .unwrap_or(Ok(RequestTimeoutOption::ClientConfig)),
+        b"WAIT" => get_timeout_from_cmd_arg(cmd, 2, TimeUnit::Milliseconds),
```

Comment:
No such thing as 'indefinitely', because the client has a default value. `0` is the same as using the client default timeout.  
Change to return `Ok(RequestTimeoutOption::ClientConfig)` when the value is 0.